src/crimson/os/seastore/onode_manager/staged-fltree/tree.h

```diff
@@ -189,10 +189,10 @@ class Btree {
   eagain_ifuture<bool> contains(Transaction& t, const ghobject_t& obj) {
     return seastar::do_with(
       full_key_t<KeyT::HOBJ>(obj),
-      [this, &t](auto& key) -> eagain_ifuture<bool> {
-        return get_root(t).si_then([this, &t, &key](auto root) {
+      [this, &t, &obj](auto& key) -> eagain_ifuture<bool> {
+        return get_root(t).si_then([this, &t, &key, &obj](auto root) {
```

Comment:
`obj` is not supposed to have extented lifecycle beyond async operations, `key` is. So the hint need to be generated from `key`.